### PR TITLE
Improve AggregationPopover component

### PR DESCRIPTION
**Pull request by gusaiani**
The `render` method here is huge and confusing.

The main outcome of this PR is extracting functions to get most of what `render` out of render itself.

Commits should be helpful in understanding changes topic by topic.

### CODE REVIEW

File: frontend/src/metabase/query_builder/components/AggregationPopover/AggregationPopover.jsx [hunk 1]
```diff
@@ -106,26 +106,26 @@ export default class AggregationPopover extends Component {
...
-    } else if (item.aggregation && item.aggregation.requiresField) {
+    } else if (item.aggregation?.requiresField) {
       // check if this aggregation requires a field, if so then force user to pick that now, otherwise we are done
       this.setState({
         aggregation: item.value,
         choosingField: true,
       });
     } else {
-      // this includse picking a METRIC or picking an aggregation which doesn't require a field
+      // this includes picking a METRIC or picking an aggregation which doesn't require a field
```
Comment (WiNloSt) [inline frontend/src/metabase/query_builder/components/AggregationPopover/AggregationPopover.jsx:128]: Nice catch.

File: frontend/src/metabase/query_builder/components/AggregationPopover/AggregationPopover.jsx [hunk 2]
```diff
@@ -147,9 +147,9 @@ export default class AggregationPopover extends Component {
     const { aggregationOperators, query, dimension, showRawData } = this.props;
     return (
       aggregationOperators ||
-      (dimension && dimension.aggregationOperators()) ||
+      dimension?.aggregationOperators() ||
       query.table().aggregationOperators()
-    ).filter(agg => showRawData || agg.short !== "rows");
+    ).filter(aggregation => showRawData || aggregation.short !== "rows");
```
Comment (WiNloSt) [inline frontend/src/metabase/query_builder/components/AggregationPopover/AggregationPopover.jsx:152]: I'm not sure, but I think `aggregation` and `aggregationOperator` would have different meanings.

I think this is the `aggregationOperator` that this code refers to

https://github.com/metabase/metabase/blob/ea1d0bda76d126904e3b6ee456d0de64a003f4ef/frontend/src/metabase/lib/schema_metadata.js#L599-L608

File: frontend/src/metabase/query_builder/components/AggregationPopover/AggregationPopover.jsx [hunk 3]
```diff
@@ -172,77 +172,26 @@ export default class AggregationPopover extends Component {
...
-        )
-      : [];
-    const metricItems = metrics.map(metric => ({
-      name: metric.name,
-      value: ["metric", metric.id],
-      isSelected: aggregation =>
-        AGGREGATION.getMetric(aggregation) === metric.id,
-      metric: metric,
-    }));
+    const maybeOveriddenShowCustomProp =
```
Comment (WiNloSt) [inline frontend/src/metabase/query_builder/components/AggregationPopover/AggregationPopover.jsx:190]: nit: typo 
```suggestion
    const maybeOverriddenShowCustomProp =
```

File: frontend/src/metabase/query_builder/components/AggregationPopover/AggregationPopover.jsx [hunk 4]
```diff
@@ -295,6 +247,82 @@ export default class AggregationPopover extends Component {
...
+      isSelected: agg =>
+        AGGREGATION.isStandard(agg) &&
+        AGGREGATION.getOperator(agg) === aggregation.short,
+      aggregation: aggregation,
+    }));
+  }
+
+  getMetrics(table, selectedAggregation) {
+    const { dimension, showMetrics } = this.props;
+    const maybeOveriddenShowMetrics = dimension ? false : showMetrics;
```
Comment (WiNloSt) [inline frontend/src/metabase/query_builder/components/AggregationPopover/AggregationPopover.jsx:285]: nit: typo 
```suggestion
    const maybeOverriddenShowMetrics = dimension ? false : showMetrics;
```

**Review by WiNloSt — APPROVED** [review]
Looks good to me! This should make the `render` function easier to understand.

File: frontend/src/metabase/query_builder/components/AggregationPopover/AggregationPopover.jsx (hunk 2 above)
Comment (gusaiani) [reply frontend/src/metabase/query_builder/components/AggregationPopover/AggregationPopover.jsx:152]: Awesome, thank you, updated.

File: frontend/src/metabase/query_builder/components/AggregationPopover/AggregationPopover.jsx (hunk 1 above)
Comment (mazameli) [reply frontend/src/metabase/query_builder/components/AggregationPopover/AggregationPopover.jsx:128]: You need [more of these](https://www.google.com/search?q=sf+giants+gifs&sxsrf=ALiCzsbkHsElYghRaNOIXEu4sRjnBKmpJw:1661816768996&source=lnms&tbm=isch&sa=X&ved=2ahUKEwinmsrxne35AhV7ATQIHQR3DesQ_AUoAXoECAEQAw&biw=1855&bih=1211&dpr=2), @WiNloSt ;)

**Pull request merged**